ceci/stage.py

```diff
@@ -18,33 +18,42 @@ class PipelineStage:
     """
     parallel = True
     config_options = {}
+    doc=""
 
     def __init__(self, args):
         args = vars(args)
         self._inputs = {}
         self._outputs = {}
+        self._configs = {}
         missing_inputs = []
         missing_outputs = []
+        missing_configs =[]
+        # if the input contains a configuration file, we don't raise the alarm
+        # just yet and will try to read missing configuration from file
+        if 'config' not in self.input_tags():
```

Comment:
I'm a bit wary of this change - it seems to blur the line between input file and configuration file.

I was thinking of the read_config method etc as being an optional mechanism, not something required.  Though I do see the advantage of confirming in advance that all required config options are specified.

Reply:
the read_config is still sort of optional, it will only be called if there is a config file in the inputs (unless I missed something, the config file is considered as an input for the pipeline elements (it's in input_tags) ). 
Otherwise, there is only a problem if there is an option that was left with default None and not provided on the command line.
Under normal operation (within ceci) this shouldn't impact the behaviour of the pipeline